spring-cloud-generator/scripts/generate-steps.sh

```diff
@@ -27,14 +27,22 @@ function compute_monorepo_version() {
 }
 
 # Generate list of in-scope libraries to use (as static data source to generate modules from)
-# Uses heuristic approach to parse googleapis commitish and other metadata from google-cloud-java
-# See generate-library-list.sh and https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/1390 for details
+# Get googleapis committish used for specified release and assign to global GOOGLEAPIS_COMMITTISH.
+# For each library in google-cloud-java, parse information from ".repo-metadata.json"
+# and hermetic build config file: generation_config.yaml
 # Assumes current working directory is spring-cloud-generator
 #
 # $1 - Monorepo version tag (or committish)
 function generate_libraries_list(){
   monorepo_commitish=$1
+  git clone https://github.com/googleapis/google-cloud-java.git
```

Comment:
nit: we could use something like sparse_checkout ([example](https://github.com/googleapis/sdk-platform-java/blob/56775a61a054fb67a15023481cd3d0a0cf2b0496/library_generation/utils/utilities.sh#L313-L327)) to speed this up

Reply:
`sparse-checkout` may not help as much in this case, because in the following steps, I need to loop in each module and look for the repo-metadata.json. However, doing a shallow clone with only 1 or 2 commit should speed things up noticeably.